tests/storage/storage.go

```diff
@@ -548,7 +548,14 @@ var _ = SIGDescribe("Storage", func() {
 		Context("[Serial]With feature gates disabled for", Serial, func() {
 			It("[test_id:4620]HostDisk, it should fail to start a VMI", func() {
 				tests.DisableFeatureGate(virtconfig.HostDiskGate)
-				vmi = tests.NewRandomVMIWithHostDisk("somepath", v1.HostDiskExistsOrCreate, "")
+				vmi = libvmi.New(
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithResourceMemory(libvmi.CirrosMemory()),
+					libvmi.WithHostDisk("host-disk", "somepath", v1.HostDiskExistsOrCreate),
+					// hostdisk needs a privileged namespace
+					libvmi.WithNamespace(testsuite.NamespacePrivileged),
```

Comment:
This is not needed, the namespace specified as part of the creation should be enough (line 560).
Is there some other magic that happens here which I missed?

This should be true to all occurences of this in this PR.

BTW, the addition of this option to libvmi is probably incorrect, it needs to be removed.
I think there was only one case when it was needed to be added to the spec, due to some webhook that depended on it (which is a bug IMO).

Reply:
This is not something I invented really, I got the same comments in the past from others and I think it is improper to specify the namespace as part of the spec.

If the NS specified by the client command collides with the one in the spec, it will blow up.
I do not see a reason to get into that scenario.
The fact the parameter can be provided from two sources is a problem and I would prefer not to encourage it if possible.

The helper you mentioned is using `testsuite.GetTestNamespace` which has the needed logic to work without the NS specified in the spec. BYW, I do not like that helper as well :)

Reply:
For this refactor I will leave things as is. We can talk about a new refactor maybe in the future, WDYT?

Reply:
I think you can keep setting the namespace here for a while, but you should not add 
`libvmi.WithNamespace` so as not to confuse others to use it more.

Reply:
Oh, I was not aware that `libvmi.WithNamespace` was introduced back in https://github.com/kubevirt/kubevirt/pull/8649